Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/SwapChainDescriptor.h

```diff
@@ -18,6 +18,7 @@ namespace AZ
     namespace RHI
     {
         using WindowHandle = Handle<uint64_t, class Window>;
+        using NativeConnection = void*;
```

Comment:
Maybe we could use a struct WindowHandle on Linux so we don't need a nativeConnection parameter on all those functions below.

Reply:
I agree. Can we do that for this PR?  Lets call it WindowDescriptor.

Reply:
or maybe even hijack the WindowHandle for NativeConnection. That may be easier.

Reply:
Since it's a void*, you can just point it to a std::pair or your own struct that contains the native handle and the connection. You just need to make sure that whoever is using knows how to cast it back.
Adding the nativeconnection as a separate member is polluting a lot of the APIs.

Reply:
Actually, in my local branch where i'm trying to get the editor running, I'm not introducing an opaque NativeConnection type and passing it around. The approach I took was to create an EBus that gets the current xcb connection, and the ebus handler is in the 'ApplicationLinux' class, where the connection is created when the object is created (application start) and disconnected on shutdown. This would also eliminate the need to pass in the connection in a lot of the calls in this PR (instead, retrieving the xcb connection through the ebus call instead)

Reply:
> Actually, in my local branch where i'm trying to get the editor running, I'm not introducing an opaque NativeConnection type and passing it around. The approach I took was to create an EBus that gets the current xcb connection, and the ebus handler is in the 'ApplicationLinux' class, where the connection is created when the object is created (application start) and disconnected on shutdown. This would also eliminate the need to pass in the connection in a lot of the calls in this PR (instead, retrieving the xcb connection through the ebus call instead)

I think this is a good solution unless you can have multiple xcb conections with the same window?